server/main_server/src/main.cpp

```diff
@@ -0,0 +1,1342 @@
+                // flushing stdio buffers and handlers registered by the parent.
+                //
+                // In the case of C++, this is necessary to avoid triggering destructors. Triggering a destructor
+                // could result in assertions made by the struct/class being violatied. For some data types,
+                // violating an assertion results in program termination (i.e. SIGABRT).
+                _exit(1);
+            }
+            else if (g_pid_ds > 0) {
+                log_server::info("{}: Delay Server PID = [{}].", __func__, g_pid_ds);
+            }
+            else {
+                log_server::error("{}: Could not launch delay server [errno={}].", __func__, errno);
+                g_pid_ds = 0;
+            }
+        }
+    } // launch_delay_server
+
+    // NOLINTNEXTLINE(readability-function-cognitive-complexity)
+    auto migrate_and_launch_delay_server(bool _write_to_stdout,
+                                         bool _enable_test_mode,
+                                         std::chrono::steady_clock::time_point& _time_start) -> void
+    {
+        if (0 == g_pid_ds) {
+            log_server::debug("{}: Checking if Agent Factory is ready to accept client requests before launching the Delay Server.", __func__);
+
+            // Defer the launch of the delay server if the agent factory isn't listening.
+            try {
+                // The server assumes the service account's irods_environment.json defines the true identity of
+                // the local server. We cannot use localhost or the loopback address because the computer may have
+                // multiple network interfaces and/or hostnames which map to different IPs.
+                const auto local_server_host = irods::get_environment_property<std::string>(irods::KW_CFG_IRODS_HOST);
+
+                // Use the zone port property from server_config.json. This property defines the port for
+                // server-to-server connections within the zone.
+                const auto local_server_port = irods::get_server_property<int>(irods::KW_CFG_ZONE_PORT);
+                const auto local_server_port_string = std::to_string(local_server_port);
+
+                if (is_server_listening_for_connections(local_server_host, local_server_port_string) != 0) {
+                    return;
+                }
+            }
+            catch (const std::exception& e) {
+                log_server::debug("{}: Connect Error: {}. Deferring launch of Delay Server.", __func__, e.what());
+                return;
+            }
+        }
+
+        // TODO Update irods/irods_docs - made /advanced_settings/migrate_delay_server_sleep_time_in_seconds a required property.
+        const auto migration_sleep_time_in_seconds = irods::get_advanced_setting<int>(irods::KW_CFG_MIGRATE_DELAY_SERVER_SLEEP_TIME_IN_SECONDS);
+        const auto now = std::chrono::steady_clock::now();
```

Comment:
Open issue for this.

Reply:
Or use a default so the property isn't required.

No issue needed.

Reply:
Restored support for allowing the delay server migration sleep time to be optionally defined.

Reply:
but... we want the config to convey the full range of what is available to the admin...

so we made it required, and populated it on new install with a default, and on upgrade if it wasn't defined.

Reply:
That option isn't required by the server.

If it's defined, we use it. If it's not, we use a default of 5 seconds.

We can of course change that behavior if preferred.

Reply:
i think it wasn't required because we wanted to introduce it, but not change the schema version.   if we're changing the schema version for 5.0, then we can make all these new things required (if we want).

Reply:
Yes, that's correct.

So the question is... Should it be required in iRODS 5?

Reply:
I think our schemas are most helpful if they require nearly everything - since we cannot ship config files with comments (b/c JSON).   And forces upgrades to get their configs 'up to speed'.

Reply:
Ok. Will make it required.

Reply:
Created #8011.

Done.